src/python/zquantum/core/graph.py

```diff
@@ -185,34 +194,42 @@ def generate_graph_from_specs(graph_specs: dict) -> nx.Graph:
     Returns:
         A networkx.Graph object
     """
-    type_graph = graph_specs['type_graph']
-    num_nodes = graph_specs['num_nodes']
-    random_weights = graph_specs.get('random_weights', False)
-    seed = graph_specs.get('seed', None)
-    number_of_cliques = graph_specs.get('number_of_cliques', None)
-    size_of_cliques = graph_specs.get('size_of_cliques', None)
-    length_of_ladder = graph_specs.get('length_of_ladder', None)
-    number_of_vertices_complete_graph = graph_specs.get('number_of_vertices_complete_graph', None)
-
-    if type_graph == 'erdos_renyi':
-        probability = graph_specs['probability']
-        graph = generate_random_graph_erdos_renyi(num_nodes, probability, random_weights, seed)
-
-    elif type_graph == 'regular':
-        degree = graph_specs['degree']
+    type_graph = graph_specs["type_graph"]
+    num_nodes = graph_specs.get("num_nodes", None)
+    random_weights = graph_specs.get("random_weights", False)
+    seed = graph_specs.get("seed", None)
+    number_of_cliques = graph_specs.get("number_of_cliques", None)
+    size_of_cliques = graph_specs.get("size_of_cliques", None)
+    length_of_ladder = graph_specs.get("length_of_ladder", None)
+    number_of_vertices_complete_graph = graph_specs.get(
+        "number_of_vertices_complete_graph", None
+    )
```

Comment:
Second argument  to `dict.get()` (the default returned value) defaults to `None` if you don't pass it. WDYT about removing it from these couple of lines?
```suggestion
    type_graph = graph_specs["type_graph"]
    num_nodes = graph_specs.get("num_nodes")
    random_weights = graph_specs.get("random_weights", False)
    seed = graph_specs.get("seed")
    number_of_cliques = graph_specs.get("number_of_cliques")
    size_of_cliques = graph_specs.get("size_of_cliques")
    length_of_ladder = graph_specs.get("length_of_ladder")
    number_of_vertices_complete_graph = graph_specs.get(
        "number_of_vertices_complete_graph"
    )
```